src/firebolt/model/V2/engine.py

```diff
@@ -194,6 +196,9 @@ def update(
             # Nothing to be updated
             return self
 
+        if name is not None and any(x is not None for x in (scale, spec, auto_stop)):
```

Comment:
Would `if name and (scale or spec or auto_stop)` work here? Or do we want to explicitly allow empty strings?

Reply:
I feel like we should explicitly only consider a parameter not set in case it's None. In case it's empty, it might be an issue on a user end, they might have an empty string in the parameter value and pass it, and it will not get updated which will be confusing. An in current case, they will get an error for the empty string